### ISSUE TITLE
Binding keys to tmux program commands fails silently

### ISSUE THREAD

**Issue by bugeats** [jwilm/alacritty#820]
I'm trying to migrate my existing iTerm2 workflow over to Alacritty + tmux. I would like to preserve high level macOS commands like Command + 1 to select the tmux "tab".

As you can see below, I can confirm that Alacritty is sending the `say` command ok. However, I can't make it work with `tmux` commands. I have confirmed that I can successfully send tmux commands from another terminal emulator -- so I'm assuming that Alacritty is dropping the ball somewhere. 

```yaml
key_bindings:
  # This works as expected
  - { key: Key3, mods: Command, command: { program: "say", args: ["one", "two", "buckle", "my", "shoe"] } }
  - { key: Key4, mods: Command, command: { program: "say", args: ["'one two buckle my shoe'"] } }

  # This does not appear to send tmux commands
  - { key: Key1, mods: Command, command: { program: "tmux", args: ["select-window", "-t", "1"] } }
  - { key: Key2, mods: Command, command: { program: "tmux", args: ["select-window", "-t", "2"] } }

  # Neither to these alternative formats:
  - { key: Key1, mods: Command, command: { program: "tmux select-window -t 1", args: [] }}
  - { key: Key2, mods: Command, command: { program: "tmux", args: ["select-window -t 2"] }}
```

macOS 10.12.6

BTW, is there a way I can debug this sort of thing? I'm not finding anything in the system console logs.

**Comment by jwilm** [comment]
I suspect the problem you're running into is that those `tmux` commands are not a subprocess of a `tmux` server. I'm not sure the best way to solve your problem. You might consider asking on the `tmux` tracker or pinging them from here.

**Comment by jwilm** [comment]
Scratch that, it's not about subprocess; I believe the commands look at the `TMUX` environment variable. That's typically only set when you're inside `tmux`. It appears to contain the socket path, session name, and server PID. There's no way right now to set environment variables for execing a key binding let alone figure out the correct tmux server pid or session name.

**Comment by bugeats** [comment]
I was able to send tmux commands from a separate iTerm2 instance and see them honored in my Alacritty instance. I don’t understand at which point the env vars would be set in iTerm2.

**Comment by jwilm** [comment]
Are you using iTerm2's tmux integration?

**Comment by bugeats** [comment]
No. tmux install is relatively fresh too. I installed in order to migrate my workflow to Alacritty. 

I just tried running `tmux select-window -t 3` from Terminal.app and it worked as expected -- the tmux client in Alacritty responded. My Terminal.app has literally never run tmux ever before.

**Comment by bugeats** [comment]
Can you provide more information about the context that programs are actually executed in? What are the env vars? I'm trying to run a bash script that's in my $PATH but it's failing. I also tried setting `program:` to be the absolute path to the script - no go.

**Comment by jwilm** [comment]
I believe programs are executed with the same environment as Alacritty. You could verify this by running a command like `/bin/bash -c "env | tee /tmp/alacritty_env` and inspecting the results.

If you want to mess around with command execution, the code can be found under [`Action::execute`](https://github.com/jwilm/alacritty/blob/68cca6eb381f88e861439ed6a1dda1ab6a7429d3/src/input.rs#L192)

**Comment by bugeats** [comment]
This is what I had to do to get it working:

https://github.com/bugeats/dotfiles/commit/ccfa4650e9db45f2d8db7f9ac416a6811847b137#diff-6983fec7bc39f877c984bd537efb8f0fR179

That `derp` script has calls to tmux which of course don't work right. 

Thanks for the code hints. I'm brand new to Rust, but I'll take a look.

**Comment by snoblenet** [comment]
It'd be great if there was a wiki page providing all the necessary detail about how to remap keys.
My first experiment is doing nothing when I press `⌘-E`:
```
  - { key: E,        mods: Command, command: { program: "echo", args: ["echo"] } }
```

**Comment by wisonye** [comment]
@bugeats My config works very well actually:

```
  # Use `<cmd + 1~9>` to switch tmux window
  - { key: Key1,      mods: Command,  command: { program: "/usr/local/bin/tmux", args: ["select-window", "-t 0"] } }
  - { key: Key2,      mods: Command,  command: { program: "/usr/local/bin/tmux", args: ["select-window", "-t 1"] } }
  - { key: Key3,      mods: Command,  command: { program: "/usr/local/bin/tmux", args: ["select-window", "-t 2"] } }
  - { key: Key4,      mods: Command,  command: { program: "/usr/local/bin/tmux", args: ["select-window", "-t 3"] } }
  - { key: Key5,      mods: Command,  command: { program: "/usr/local/bin/tmux", args: ["select-window", "-t 4"] } }
  - { key: Key6,      mods: Command,  command: { program: "/usr/local/bin/tmux", args: ["select-window", "-t 5"] } }
  - { key: Key7,      mods: Command,  command: { program: "/usr/local/bin/tmux", args: ["select-window", "-t 6"] } }
  - { key: Key8,      mods: Command,  command: { program: "/usr/local/bin/tmux", args: ["select-window", "-t 7"] } }
  - { key: Key9,      mods: Command,  command: { program: "/usr/local/bin/tmux", args: ["select-window", "-t 8"] } }
```
